### DISCUSSION TITLE
Using Strawberry Perl for Oracle-Access

### DISCUSSION THREAD

**Discussion by TheOncomingStorm57** [StrawberryPerl/Perl-Dist-Strawberry#216] [Q&A]
Dear Community,

I enjoy using Strawberry Perl scripts for Oracle database evaluations. 
Unfortunately, DBD support has been missing since strawberry-perl-5.32.1.1. 
Is it correct that it will not return with strawberry-perl-5.40.0.1 either? 
Will it be gone permanently?

**Comment by genio** [comment]
DBD and DBI are still there. What method are you trying and what's not working?

**Comment by TheOncomingStorm57** [comment]
I use something like that:
### Code
`#! /usr/local/bin/perl
use DBI;
   $data_source   = $ARGV[0];
   $username      = $ARGV[1];
   $auth          = $ARGV[2];
   $delim = "|";

$drh = DBI->install_driver('Oracle') or die "Error:install_driver Keine Verbindung zur Datenbanktreiber:\n$DBI::errstr\n"; 
$db = $drh->connect($data_source, $username, $auth) or die"Error: connect Keine Verbindung zur Datenbank:\n$DBI::errstr\n";

$stmt="SELECT DISTINCT u.TABLE_NAME  FROM user_tab_columns u WHERE u.TABLE_NAME like '%BEARBN%'  ORDER BY u.TABLE_NAME";
$db->{RaiseError} = 1;
print "prepare Test-Statement \n";
eval {
	$st = $db->prepare($stmt);
};
if ($@) {
   print "Fatal error: ";
   $db->errstr, "\n";
} else {
   print "Prepare is OK $@\n";
};

print "Test-Statement \'$stmt\'\n";
$st->execute or die "Fehler !!! Error: $DBI::errstr\n$stmt\n";
my $record;
while (defined($record = $st->fetch())) {
	$i++;
   print "result".$i."==>".join($delim, @{$record}), "\n";
}
$st->finish();

exit(0);
`
### Problem: 
install_driver(Oracle) failed: Can't locate DBD/Oracle.pm in @INC (you may need to install the DBD::Oracle module) (@INC entries checked: C:/DEV/strawberry-perl-5.40.0.1-64bit-portable/perl/site/lib C:/DEV/strawberry-perl-5.40.0.1-64bit-portable/perl/vendor/lib C:/DEV/strawberry-perl-5.40.0.1-64bit-portable/perl/lib) at (eval 10) line 3.
Perhaps the DBD::Oracle perl module hasn't been fully installed,
or perhaps the capitalisation of 'Oracle' isn't right.
Available drivers: ADO, CSV, DBM, ExampleP, File, Gofer, Mem, ODBC, Pg, Proxy, SQLite, Sponge.
 at C:\test_dbi.pl line 10.

### my expectation:
prepare Test-Statement
Prepare is OK
Test-Statement 'SELECT DISTINCT u.TABLE_NAME  FROM user_tab_columns u WHERE u.TABLE_NAME like '%BEARBN%'  ORDER BY u.TABLE_NAME'
result1==>tableA
result2==>tableB
result3==>tableC

**Comment by genio** [comment]
https://github.com/StrawberryPerl/Perl-Dist-Strawberry/issues/69

**Comment by xiaoyafeng** [comment]
Since Oracle 19, Oracle provide a image type of client which include a DBD with a perl. It's portable, free and easy-use. you could download ,extract and use it(perl, DBI and DBD::Oracle) directly,  you also could  use it in strawberry perl by use lib pragma.
HTH.

**Reply by TheOncomingStorm57** [reply]
Thank you for this suggestion. I will try it out. 
However, it would be nicer if it worked directly with the Strawberry installer again. ( .."like In the good old days..." 😊 )
In my company, the software is distributed centrally by the IT department. They are very reluctant to enrich the installer from other sources before distribution.

**Reply by TheOncomingStorm57** [reply]
Greetings, I have supplemented the code from above with the Path from my Oracle installation

use lib 'C:/xxx/OracleClient193000/perl/site/lib';
use DBI;

still Problem:
DBI object version 1.643 does not match bootstrap parameter 1.642 at C:/DEV/perl/strawberry-perl-5.40.0.1-64bit-portable/perl/lib/DynaLoader.pm line 223.
BEGIN failed--compilation aborted at C:/DevSL/OracleClient193000/perl/site/lib/DBI.pm line 284.
Compilation failed in require at test_dbi.pl line 6.
BEGIN failed--compilation aborted at test_dbi.pl line 6.

Can you help me ?

**Reply by shawnlaffan** [reply]
That suggests mixed versions of DBI.  There is probably also one under `.../OracleClient193000/perl/site/lib`.

See also https://www.perlmonks.org/?node_id=1158036

**Reply by TheOncomingStorm57** [reply]
Unfortunately, ti have still problems with mixed versions of DBI. I already have an Oracle client installed by IT. 
I am trying to access DBI with a portable Perl 5.40. 
Can this work without a new installation of the Oracle and Perl clients? As genio wrote on September 11, the Oracle access was removed in the Strawberry installer (#69 ). 
I have to migrate to Strawberry 5.40 because of #229 , as soon as ist is possible. But it doesn’t work due to the missing Oracle access..  :-(

**Reply by shawnlaffan** [reply]
Have you tried installing `DBD::Oracle` using Strawberry Perl 5.40?  

It needs some paths and env vars to be set before doing so.  The SP 5.32 build recipe could probably be adapted: https://github.com/StrawberryPerl/Perl-Dist-Strawberry/blob/8ec82760ac883c7a6a00425fc2ec53967526b528/share/64bit-5.32.1.1.pp#L203
